concordium-smart-contract-testing/src/lib.rs

```diff
@@ -0,0 +1,2745 @@
+            let number_of_sigs = 1; // Accounts always have one signature here. TODO: Should we allow changing that?
+            let base_cost = cost::base_cost(payload_size, number_of_sigs);
+            let deploy_module_cost = cost::deploy_module(payload_size);
+            let total = base_cost + deploy_module_cost;
+            println!(
+                "Deploying module \
+                 cost:\n\tmodule_size:{}\n\tbase_cost:{}\n\tdeploy_module_cost:{}\n\ttotal:{}",
+                payload_size, base_cost, deploy_module_cost, total
+            );
+            total
+        };
+        let transaction_fee = self.calculate_energy_cost(energy);
+        println!(
+            "Deploying module with size {}, resulting in {} NRG.",
+            module.len(),
+            energy
+        );
+
+        // Try to subtract cost for account
+        match self.accounts.entry(sender) {
+            Vacant(_) => return Err(DeployModuleError::AccountDoesNotExist),
+            Occupied(mut acc) => {
+                let acc = acc.get_mut();
+                if acc.balance < transaction_fee {
+                    return Err(DeployModuleError::InsufficientFunds);
+                }
+                acc.balance -= transaction_fee;
+            }
+        }
+
+        // Save the module
+        let module_reference = {
+            let mut hasher = Sha256::new();
+            hasher.update(module);
+            let hash: [u8; 32] = hasher.finalize().into();
+            ModuleReference::from(hash)
+        };
+        self.modules.insert(module_reference, artifact);
+        Ok(SuccessfulModuleDeployment {
+            module_reference,
+            energy,
+            transaction_fee,
+        })
+    }
+
+    /// Deploy a raw wasm module, i.e. one **without** the prefix of 4 version
+    /// bytes and 4 module length bytes.
+    pub fn module_deploy_raw<P: AsRef<Path>>(
+        &mut self,
+        sender: AccountAddress,
+        module_path: P,
+    ) -> Result<SuccessfulModuleDeployment, DeployModuleError> {
+        // Load file
+        let module = std::fs::read(module_path)?;
+        self.module_deploy_aux(sender, &module)
+    }
+
+    /// Deploy a wasm module as it is output from `cargo concordium build`, i.e.
+    /// **including** the prefix of 4 version bytes and 4 module length bytes.
+    pub fn module_deploy<P: AsRef<Path>>(
+        &mut self,
+        sender: AccountAddress,
+        module_path: P,
+    ) -> Result<SuccessfulModuleDeployment, DeployModuleError> {
+        // Load file
+        let module = std::fs::read(module_path)?;
+        // Here, we skip the 8 bytes that encode the smart contract version and module
+        // length
+        self.module_deploy_aux(sender, &module[8..])
+    }
+
+    pub fn contract_init(
+        &mut self,
+        sender: AccountAddress,
+        module_reference: ModuleReference,
+        contract_name: ContractName,
+        parameter: ContractParameter,
+        amount: Amount,
```

Comment:
I do have some reasons for not using that.
1. It uses `ModuleRef` but we use `ModuleReference` everywhere else here
2. It is slightly easier to create `ContractName` than the owned version:
    - `ContractName::new_unchecked("init_counter")`
    -  `OwnedContractName::new_unchecked("init_counter".into())`
 3. The `ContractParameter` type has a few helper methods that are easier to use. In particular, `from_typed`

Reply:
The points in my comment above are not that so relevant or important anymore, but I still think it is easier to call a function a provide parameters directly instead of inside a struct.